sw/device/silicon_creator/mask_rom/primitive_bootstrap.c

```diff
@@ -89,19 +92,49 @@ static rom_error_t bootstrap_requested(bool *result) {
   return kErrorOk;
 }
 
+/**
+ * Check that flash data partitions are all blank.
+ */
+bool flash_is_empty(void) {
+  dif_flash_ctrl_device_info_t flash_info = dif_flash_ctrl_get_device_info();
+  const uint32_t flash_size_bytes =
+      flash_info.num_banks * flash_info.bytes_per_page * flash_info.data_pages;
+  uint32_t *const limit =
+      (uint32_t *)(TOP_EARLGREY_FLASH_CTRL_MEM_BASE_ADDR + flash_size_bytes);
+  uint32_t mask = -1u;
+  uint32_t *p = (uint32_t *)TOP_EARLGREY_FLASH_CTRL_MEM_BASE_ADDR;
+  for (; p < limit;) {
+    mask &= *p++;
+    mask &= *p++;
+    mask &= *p++;
+    mask &= *p++;
+    mask &= *p++;
+    mask &= *p++;
+    mask &= *p++;
+    mask &= *p++;
+    if (mask != -1u) {
+      return false;
+    }
+  }
+  return true;
+}
+
 /**
  * Erase all flash, and verify blank.
  */
-static rom_error_t erase_flash(void) {
-  if (flash_bank_erase(FLASH_BANK_0) != 0) {
+static int erase_flash(void) {
```

Comment:
same here, testutils?

Reply:
These use error codes that are specific to the implementation.